array_api_tests/test_creation_functions.py

```diff
@@ -130,9 +147,32 @@ def test_full(shape, fill_value, dtype):
     else:
         assert all(equal(a, asarray(fill_value, **kwargs))), "full() array did not equal the fill value"
 
-# TODO: implement full_like (requires hypothesis arrays support)
-def test_full_like():
-    pass
+
+@composite
+def fill_values(draw):
+    kw = draw(shared(kwargs(dtype=none() | xps.scalar_dtypes()), key="full_like_kw"))
+    dtype = kw.get("dtype", None) or draw(shared_dtypes)
+    return draw(xps.from_dtype(dtype))
+
+
+@given(
+    x=xps.arrays(dtype=shared_dtypes, shape=shapes),
+    fill_value=fill_values(),
+    kw=shared(kwargs(dtype=none() | xps.scalar_dtypes()), key="full_like_kw"),
+)
+def test_full_like(x, fill_value, kw):
+    out = full_like(x, fill_value, **kw)
+    dtype = kw.get("dtype", None) or x.dtype
+    if kw.get("dtype", None) is None:
+        assert out.dtype == x.dtype, f"{x.dtype=!s}, but full_like() returned an array with dtype {out.dtype}"
+    else:
+        assert out.dtype == dtype, f"{dtype=!s}, but full_like() returned an array with dtype {out.dtype}"
+    assert out.shape == x.shape, "{x.shape=}, but full_like() returned an array with shape {out.shape}"
+    if is_float_dtype(dtype) and isnan(asarray(fill_value)):
```

Comment:
fill_value will be a Python float so we can just check `math.isnan`. It's better to not make tests depend on other API functions when it's not necessary.